pos_barcode_sync/static/src/js/pos_barcode_sync.js

```diff
@@ -0,0 +1,45 @@
+odoo.define('pos_barcode_sync.pos', function (require) {
+
+    var session = require('web.session');
+    var Backbone = window.Backbone;
+    var core = require('web.core');
+    var screens = require('point_of_sale.screens');
+    var models = require('point_of_sale.models');
+    var longpolling = require('pos_longpolling');
+    var gui = require('point_of_sale.gui');
+
+    var _t = core._t;
+
+    var PosModelSuper = models.PosModel;
+    models.PosModel = models.PosModel.extend({
+        initialize: function(){
+            PosModelSuper.prototype.initialize.apply(this, arguments);
+            this.bus.add_channel_callback("pos_barcode_sync", this.on_barcode_updates, this);
+        },
+        on_barcode_updates: function(data){
+            var self = this;
+            PosModelSuper.prototype.load_new_partners.apply(this, arguments).done(function(){
+                var client_list_screen = self.gui.screen_instances.clientlist;
+                client_list_screen.render_list(self.db.get_partners_sorted(1000));
+                if (client_list_screen){
+                    client_list_screen.update_opened_partner_details(data.partner_ids);
+                }
+            });
+        },
+    });
+
+    gui.Gui.prototype.screen_classes.filter(function(el) {
+        return el.name === 'clientlist';
+    })[0].widget.include({
+        update_opened_partner_details: function(partner_ids){
+            if (this.$el.not('.oe_hidden')[0]) {
```

Comment:
It checks for any hidden element. Check main block only. Consult with @GabbasovDinar

Reply:
@KolushovAlexandr var current_screen_name = this.pos.gui.get_current_screen();